engine.py

```diff
@@ -912,3 +912,58 @@ def __setup_favorite_dirs(self):
                                 type=(nuke.IMAGE | nuke.SCRIPT | nuke.GEO),
                                 icon=icon_path,
                                 tooltip=path)
+
+    #####################################################################################
+    # Script and session related methods
+
+    def get_session_path(self):
+        """
+        Returns the path to the current file if it resides on
+        disk. If unsaved, it returns None.
+
+        :return: Path to the current scene if file is saved, else returns
+                 and empty string.
+        """
+        # retrieve the name of the nuke file. In case of a saved file, the
+        # absolute path of the file is returned, for an unsaved file,
+        # an empty string is returned.
+        return nuke.root().knob("name").value()
+
+    def get_session_dependencies(self):
+        """
+         Returns the list of file or folder dependencies for the current
+         session.
+
+         :return: A list of file or folder paths that the current
+                  session depends on.
+         """
+        dependencies = []
```

Comment:
I think we should have an engine hook that allows to filter the list of session dependencies so clients can either block certain dependencies from being returned or add ones that we may have missed.

@manneohrstrom @robblau ?

Reply:
yes absolutely! We should give a hook the ability to post process the list. This was discussed but my thinking was that we keep it simple for the time being and we can potentially add that in the future. @jfboismenu if you strongly disagree, let's discuss and see if we can add it in now. It's not a big change i think.